### PR TITLE
Update agent flow

### PR DESCRIPTION
**Pull request by stellaHSR**
**Features**

- Modify the code directory, and make it more clearly
- Update agent flow, and add a new role that inherits the Data Interpreter.
- Update run_api.py, make it more readable 
    
**Feature Docs**

**Influence**

**Result**

img_v3_0297_4f321356-ed97-4e0b-b2eb-90f45c0aa07g
img_v3_0297_c34d6bef-49ce-47a8-9306-5d1d4f60478g
img_v3_0297_930eab97-fd0d-482a-b823-a594d0105f6g

**Other**

### CODE REVIEW

File: data/inference/const.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    "matplotlib__matplotlib-24362",
+    "matplotlib__matplotlib-20584",
+    "matplotlib__matplotlib-23188",
+    "matplotlib__matplotlib-24403",
+    # 'matplotlib__matplotlib-21443',
+    # 'matplotlib__matplotlib-23047'
+]
+
+
+def read_sub_set_instance(path=SUBSET_DATASET, tag="scikit-learn"):
```
Comment (geekan) [inline data/inference/const.py:41]: sub_set -> subset

File: swe_bench/make_datasets/make_dataset.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+# @Desc    :
+import os
+from pathlib import Path
+
+from tqdm.auto import tqdm
+
+from data.inference.const import TESTBED
+from metagpt.logs import logger
+from swe_bench.make_datasets.make_instance import prompt_style_2_edits_only
+from swe_bench.utils.parse_diff import filter_changed_line
```
Comment (mannaandpoem) [inline swe_bench/make_datasets/make_dataset.py:12]: parse_diff  does not exist in swe_bench.utils

File: swe_bench/inference/run_api.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
+import json
+from pathlib import Path
+
+import fire
+from data.load_dataset import load_oracle_dataset
```
Comment (mannaandpoem) [inline swe_bench/inference/run_api.py:5]: load_dataset does not exist in data folder

**Review by geekan — APPROVED** [review]

File: swe_bench/make_datasets/make_dataset.py (hunk 2 above)
Comment (stellaHSR) [reply swe_bench/make_datasets/make_dataset.py:12]: https://github.com/geekan/MetaGPT/pull/1077

File: swe_bench/inference/run_api.py (hunk 3 above)
Comment (stellaHSR) [reply swe_bench/inference/run_api.py:5]: https://github.com/geekan/MetaGPT/pull/1077

**Pull request merged**